src/ts/interfaces.ts

```diff
@@ -17,6 +17,19 @@ interface MetadataJson {
   metadata: string;
 }
 
+interface ExtraFieldsV2 {
+  version : number;
+  data: Array<ExtraFieldsDataV2>;
+}
+
+interface ExtraFieldsDataV2 {
```

Comment:
I was thinking that having an `ExtraFields` interface that could be several versions (so a superset of the "daughter" versions) would be better: I don't like to have this V2 in the name of the thing. Also, I would define the current one as v0 and the new one as v1.

Reply:
I'm trying to implement this currently.